File: Python/3-match.py
import re

s = 'Hello user1@domain.com and user2@domain.com'
rx = r'\S+@[\S.]+'

match = re.match(rx, s)
search = re.search(rx, s)
findall = re.findall(rx, s)

print("match: ", match)
print("search: ", search.group())
print("findall: ", findall)

File: Scala/3-match.scala
package regexp

object Match extends App {
  val s = "Hello user1@domain.com and user2@domain.com"
  val rx: Regex = "\\S+@[\\S.]+".r

  val firstMatch: Option[Regex.Match] = rx.findFirstMatchIn(s)
  val allMatches: Iterator[Regex.Match] = rx.findAllMatchIn(s)

  firstMatch.map(_.group(0)).foreach(println)
  allMatches.map(_.group(0)).foreach(println)
}